components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/InMemoryBasedTemplateManager.java

```diff
@@ -0,0 +1,263 @@
+ */
+
+package org.wso2.carbon.email.mgt.store;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.email.mgt.internal.I18nMgtDataHolder;
+import org.wso2.carbon.identity.governance.exceptions.notiification.NotificationTemplateManagerServerException;
+import org.wso2.carbon.identity.governance.model.NotificationTemplate;
+import org.wso2.carbon.identity.governance.service.notification.NotificationChannels;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is responsible for storing notification templates from the email-admin-config.xml file.
+ * This class is primarily used to provide default notification templates without addition/modification,
+ * allowing for quick access of templates without migration.
+ * Templates are stored in nested maps, where the outer map is keyed by the template display name and the inner map
+ * is keyed by locale. This allows for easy retrieval of localized templates.
+ * This class used by {@link HybridTemplateManager} as a fallback mechanism.
+ * This class only supports for tenant specific notification templates.
+ */
+public class InMemoryBasedTemplateManager implements TemplatePersistenceManager {
+
+    private final Map<String, Map<String, NotificationTemplate>> defaultEmailTemplates;
+    private final Map<String, Map<String, NotificationTemplate>> defaultSMSTemplates;
+
+    /**
+     * Initializes the in-memory template manager by populating default email and SMS templates.
+     */
+    public InMemoryBasedTemplateManager() {
+
+        defaultEmailTemplates = populateTemplates(I18nMgtDataHolder.getInstance().getDefaultEmailTemplates());
+        defaultSMSTemplates = populateTemplates(I18nMgtDataHolder.getInstance().getDefaultSMSTemplates());
+    }
+
+    @Override
+    public void addNotificationTemplateType(String displayName, String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        // This method does nothing cause addition is not allowed for in-memory template manager.
+    }
+
+    /**
+     * Checks if a notification template type exists in the in-memory store.
+     *
+     * @param displayName         The display name of the template type.
+     * @param notificationChannel The channel of the notification (e.g., Email, SMS).
+     * @param tenantDomain        The tenant domain.
+     * @return true if the template type exists, false otherwise.
+     * @throws NotificationTemplateManagerServerException if an error occurs during the operation.
+     */
+    @Override
+    public boolean isNotificationTemplateTypeExists(String displayName, String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        if (!StringUtils.isBlank(displayName)) {
+            if (NotificationChannels.SMS_CHANNEL.getChannelType().equals(notificationChannel)) {
```

Comment:
why the method `getTemplateMap()` not used here?

Reply:
Ack

Reply:
Fixed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/256/commits/8628c42d5f74b1e58a734be7fc641c3f846af46d